src/app/custom-navigation-mode/custom-navigation-mode-factory.provider.ts

```diff
@@ -0,0 +1,13 @@
+import { BaseNavigationModeFactory, WizardComponent, NavigationMode } from 'angular-archwizard';
+import { CustomNavigationMode } from './custom-navigation-mode';
+
+export class CustomNavigationModeFactory extends BaseNavigationModeFactory {

```

Comment:
Should we extend the `BaseNavigationModeFactory` class inside the demo or implement the `NavigationModeFactory`?

Reply:
I think that implementing `BaseNavigationModeFactory` is more practical. Not only it is easier to implement, but also occasional library users visiting this demo page will most likely need `BaseNavigationModeFactory` rather than `NavigationModeFactory`.

By providing an interface and a base class, we make sure that users can possibly create their own implementation from scratch (implementing an interface). But vast majority of users will only need `BaseNavigationModeFactory`.